src/RA_StringUtils.cpp

```diff
@@ -145,21 +145,21 @@ void StringBuilder::AppendToString(_Inout_ std::string& sResult) const
                 break;
 
             case PendingString::Type::StringRef:
-                nNeeded += pPending.Ref.String->length();
+                nNeeded += std::get<const std::string*>(pPending.Ref)->length();
```

Comment:
With `std::variant` you can get the desired type either by index or type via `std::get`. I decided to do it by type since it's clearer that way.